docs-src/docs/guides/walkthrough.md

```diff
@@ -602,11 +649,182 @@ However, if you intend to continue through [Deployment 3](#deployment-3-guestboo
 
 ## 2b: Deploy with Terraform
 
+As before, we can complete this deployment using Terraform and the Layer0 provider instead of the Layer0 CLI. As before, we will assume that you've cloned the [layer0-examples](https://github.com/quintilesims/layer0-examples) repo and are working in the `iterative-walkthrough/deployment-2/` directory.
+
+We'll use these files to manage our deployment with Terraform:
+
+|Filename|Purpose|
+|---|---|
+|`Guestbook.Dockerrun.aws.json`|Template for running the Guestbook application|
+|`layer0.tf`|Provisions resources; populates variables in template files|
+|`Redis.Dockerrun.aws.json`|Template for running the Redis application|
+|`terraform.tfstate`|Tracks status of deployment _(created and managed by Terraform)_|
+|`terraform.tfvars`|Variables specific to the environment and application(s)|
+
+!!! Note "Note: `terraform.tfstate`"
+    You can follow the steps in this section with no `terraform.tfstate` file, and Terraform will create it and manage it as it did before. However, if you've opted **not** to destroy the resources created in [Deployment 1b](#1b-deploy-with-terraform), you can copy your previous `terraform.tfstate` file to your current working directory and Terraform will update and delete existing resources as needed in addition to creating new ones.
+
 
 ---
 
-### Part 1:
+### `Layer0.tf`: A Brief Aside: Revisited
+
+This file hasn't changed much since the first deployment - although, we _have_ added some things to it. You can see that the configurations for both the Guestbook _and_ the Redis services are contained within this single file.
+
+If you've gone through the [2a Deployment](#2a-deploy-with-layer0-cli) using the Layer0 CLI above, you'll recall that we had to obtain the URL of the Redis load balancer and manually plug it into the `Guestbook.Dockerrun.aws.json` task definition file. You may be wondering how we're supposed to do that, when Terraform creates all the resources at once. Good news! We don't have to! We can configure `layer0.tf` to get that information and plug it in automatically!
+
+When we manually edited that Guestbook task definition, we reconfigured a section that looked like this:
+
+```
+    ...
+    "environment": [
+        {
+            "name": "REDIS_ADDRESS_AND_PORT",
+            "value": "${redis_address}"
+        }
+    ],
+    ...
+```
+
+You've probably noticed that `${redis_address}` is a Terraform interpolation. If you look for the `template_file` resource labeled `guestbook` in `layer0.tf`, you'll see that in addition to supplying the Guestbook task definition file, we're also configuring a variable to be passed to the template:
```

Comment:
Maybe a link to [Interpolation Syntax](https://www.terraform.io/docs/configuration/interpolation.html) here?